sphereMove/SphereMove.cs

```diff
@@ -152,56 +200,115 @@ IEnumerator DelayedUpdate()
         // Run through interpolatePosX elements and update when the respective time is reached
         while (true)
         {
-            if (!new_trajectory)
+            switch (currentState)
             {
-                for (int i = 0; i < interpolatedPosX.Count; i++)
-                {
-                    car.x = interpolatedPosX[i];
-                    car.y = interpolatedPosY[i];
-                    car.date = interpolatedPosTime[i];
-                    Vector3 newPosition = new Vector3(car.x, car.y, this.gameObject.transform.localPosition.z);
-                    this.gameObject.transform.localPosition = newPosition;
-                    Debug.Log("Update x=" + this.gameObject.transform.localPosition.x.ToString() + "y=" + this.gameObject.transform.localPosition.y.ToString());
-
-                    // Wait until the next time to update the position
-                    yield return new WaitForSeconds((float)(interpolatedPosTime[i + 1] - interpolatedPosTime[i]).TotalSeconds);
-
-                }
-
-                // All interpolated positions have been used. Calculate new trajectory in the other coroutine
-                new_trajectory = true;
+                case State.Begin:

```

Comment:
Do we need a case for the begin state here if it is already handled above? What if we assigned the state to `Running` in the State.Begin case of the GetCarData function and then not have this case here?

Reply:
Didn't code to test but I'd say it's definitely doable as long as you initialize current_tracking_time = listTime[0] in the GetCarData routine. I split the states in DelayedUpdate just in case there will be different car tracking implementations for any of the four stages. But if it turns out that they runs the same trajectory tracking logic, then it would be good not to switch states in DelayedUpdate which repeats the same code.

Reply:
Let's leave it for now and maybe remove it in the future if we want to clean it up.